ext/iconv/config.m4

```diff
@@ -28,20 +28,11 @@ if test "$PHP_ICONV" != "no"; then
 
     if test -z "$iconv_impl_name"; then
       AC_MSG_CHECKING([if using GNU libiconv])
-      AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <iconv.h>
-#include <stdio.h>
-int main(void) {
-  printf("%d", _libiconv_version);
-  return 0;
-}
-      ]])],[
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>], [return _libiconv_version])],[
```

Comment:
```suggestion
      AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>], [int v = _libiconv_version])],[
```

Sorry, one more fix here. The test program should be still sort of properly executable with return code of 0 from the coding style point of view. Otherwise, this is only CS fix now. It would be even sufficient probably to only check it with AC_CHECK_DECL but it's fine to to compilation and link check.

Reply:
Actually, let's simply do the `(void)` here. Because of the edge case of `CFLAGS="-Werror=unused-variable"`. So it's even more compliant test program.

```suggestion
      AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>], [(void) _libiconv_version])],[
```

The -liconv linking step is done in that PHP_SETUP_ICONV where it should find some iconv library. That's ok. These further checks are only addon and sanity checks if the found iconv library can be used in PHP.

Reply:
ahhh, I see! pushed:-)